source/neuropod/bindings/c/np_tensor_allocator.h

```diff
@@ -0,0 +1,51 @@
+/* Copyright (c) 2020 UATC, LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Inspired by the TensorFlow C API
+
+#pragma once
+
+#include "neuropod/bindings/c/np_tensor.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct NP_TensorAllocator NP_TensorAllocator;
```

Comment:
I don't follow this and other declarations above for NP_Status, NP_Neuropod, NP_NeuropodTensor- is it just a stub and actual declarations will be added later?

Reply:
These are the public facing header files (so these structs are opaque to users).

The definitions will be in `_internal.h` files which will be added in another PR

See the TensorFlow C API for an [example](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/c/tf_status_internal.h) of this pattern